### PR TITLE
Memory Write resilience

### PR DESCRIPTION
**Pull request by mdurrant-b3**
As the response code changes per device/partition names, the only way to handle it is this terrible, terrible way.

### CODE REVIEW

File: acos_client/v21/action.py [hunk 1]
```diff
@@ -39,5 +39,9 @@ def reload(self, write_memory=False, **kwargs):
     def activate_and_write(self, partition, **kwargs):
         write_format = "active-partition {0}\r\nwrite memory\r\n"
         post_body = write_format.format(partition)
-
-        return self._request("POST", "cli.deploy", params=None, payload=post_body, **kwargs)
+        # Request raises an exception when the "maybe error" is returned.
+        # We can't do anything about "maybe" errors or otherwise at this stage
+        try:
+            return self._request("POST", "cli.deploy", params=None, payload=post_body, **kwargs)
+        except:
```
Comment (dougwig) [inline acos_client/v21/action.py]: Can't you at least catch ACOSException here?  Bonus points for re-raising if a string match on msg fails?

Comment (mdurrant-b3) [inline acos_client/v21/action.py]: Yeah, I had thought about that and then had the thought of  "we can't do anything about it if it does fail except notify the user".   Matching on string and re-raising if it's not the droid we're looking for seems reasonable - then we can have any error issues handled at the consumer level.

File: acos_client/v21/action.py [hunk 2]
```diff
@@ -39,5 +39,8 @@ def reload(self, write_memory=False, **kwargs):
     def activate_and_write(self, partition, **kwargs):
         write_format = "active-partition {0}\r\nwrite memory\r\n"
         post_body = write_format.format(partition)
-
-        return self._request("POST", "cli.deploy", params=None, payload=post_body, **kwargs)
+        # Request raises an exception when the "maybe error" is returned.
+        try:
+            return self._request("POST", "cli.deploy", params=None, payload=post_body, **kwargs)
+        except acos_errors.ACOSException:
```
Comment (dougwig) [inline acos_client/v21/action.py]: If this was v30 I'd want to see a further check on the string, but as it's v21, +1

**Comment by mdurrant-b3** [comment]
check

**Pull request merged**